src/sentry/api/bases/dashboard.py

```diff
@@ -24,3 +38,63 @@ def _get_dashboard(self, request, organization, dashboard_id):
             id=dashboard_id,
             organization_id=organization.id
         )
+
+
+class OrganizationDashboardWidgetEndpoint(OrganizationDashboardEndpoint):
+    def convert_args(self, request, organization_slug, dashboard_id, widget_id, *args, **kwargs):
+        args, kwargs = super(OrganizationDashboardWidgetEndpoint,
+                             self).convert_args(request, organization_slug, dashboard_id, widget_id, *args, **kwargs)
+
+        try:
+            kwargs['widget'] = self._get_widget(
+                request, kwargs['organization'], dashboard_id, widget_id)
+        except Widget.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        return (args, kwargs)
+
+    def _get_widget(self, request, organization, dashboard_id, widget_id):
+        return Widget.objects.get(
+            id=widget_id,
+            dashboard_id=dashboard_id,
+        )
+
+
+class WidgetDataSourceSerializer(serializers.Serializer):
```

Comment:
These could go in `api/serializers/rest_framework/widgets.py` (or something similar). Looks like we've put shared serializers here before.